src/lib.rs

```diff
@@ -299,6 +308,164 @@ impl Reflection {
         })
     }
 
+    fn byte_offset_to_last_var(
+        reflect: &Module,
+        struct_instruction: &Instruction,
+    ) -> Result<u32, ReflectError> {
+        debug_assert!(struct_instruction.class.opcode == spirv::Op::TypeStruct);
+
+        // if there are less then two members there is no offset to use
+        if struct_instruction.operands.len() < 2 {
+            return Ok(0);
+        }
+
+        let result_id = match struct_instruction.result_id {
+            Some(id) => id,
+            None => return Err(ReflectError::MissingResultId(struct_instruction.clone())),
+        };
+
+        let variable = reflect
+            .debugs
+            .iter()
+            .filter(|i| i.class.opcode == spirv::Op::Name)
+            .find_map(|i| {
+                if get_operand_at!(i, Operand::IdRef, 0).ok()? == result_id {
+                    Some(i)
+                } else {
+                    None
+                }
+            });
+
+        let variable = match variable {
+            Some(i) => i,
+            None => return Err(ReflectError::UnknownStruct(struct_instruction.clone())),
+        };
+
+        let idref = get_operand_at!(variable, Operand::IdRef, 0)?;
```

Comment:
Why is this finding an `OpName` for `result_id` as 0th operand, then extracting the 0th operand again? That's the same thing?

In other words, `assert_eq!(result_id, idref)`...?